department-of-veterans-affairs/caseflow
# Allow cavc user to send an ihp task to the colocated team

Bumps [CASEFLOW-107]

### Description
Allows cavc users to add an ihp admin action.

### Acceptance Criteria
- [ ] Task action "Send IHP to colocated team" is added to SendCavcRemandProcessedLetterTask
- [ ] Provide Instructions and context for this action appears on the submit admin actions page
- [ ] Creates a IhpColocatedTask assigned to the Colocated team as a child of the distribution task

### Testing Plan
1. Log in as the CAVC team admin and assign any open `SendCavcRemandProcessedLetterTask` to yourself
1. Click the task action dropdown and confirm new "send ihp to colocated team" action
1. Confirm you can add instructions to this action
1. Confirm clicking submit keeps you on the case details page and creates a privacy act task assigned to the privacy team is created as a child of the distribution task
1. Confirm you still have actions available to reassign or complete
1. Confirm you cannot create another task assigned to the colocated team
```ruby
Appeal 922 (evidence_submission)                       ID   STATUS      ASGN_BY                ASGN_TO                UPDATED_AT
└── RootTask                                           2745 on_hold                            Bva                    2020-11-16 15:55:51 UTC
    └── DistributionTask                               2746 on_hold                            Bva                    2020-11-16 15:55:51 UTC
        ├── CavcTask                                   2747 on_hold                            Bva                    2020-11-16 15:55:51 UTC
        │   └── SendCavcRemandProcessedLetterTask      2748 on_hold                            CavcLitigationSupport  2020-11-17 16:00:18 UTC
        │       └── SendCavcRemandProcessedLetterTask  3367 in_progress CAVC_LIT_SUPPORT_ADMIN CAVC_LIT_SUPPORT_ADMIN 2020-11-17 16:00:22 UTC
        └── IhpColocatedTask                           3368 on_hold     CAVC_LIT_SUPPORT_ADMIN Colocated              2020-11-17 16:04:47 UTC
            └── IhpColocatedTask                       3369 assigned    CAVC_LIT_SUPPORT_ADMIN BVALSPORER             2020-11-17 16:04:47 UTC
```

### User Facing Changes

 Workflow Step|UI
 ---|---
New task action|<img width="935" alt="Screen Shot 2020-11-17 at 11 00 34 AM" src="https://user-images.githubusercontent.com/45575454/99415989-5df67280-28c6-11eb-8238-0866e358c387.png">
Modal|<img width="468" alt="Screen Shot 2020-11-17 at 11 01 26 AM" src="https://user-images.githubusercontent.com/45575454/99415997-60f16300-28c6-11eb-8426-74333c33aef2.png">
Modal validation|<img width="462" alt="Screen Shot 2020-11-17 at 11 01 33 AM" src="https://user-images.githubusercontent.com/45575454/99416021-65b61700-28c6-11eb-93d2-b0e74f356f3d.png">
Success & updated task snapshot|<img width="1011" alt="Screen Shot 2020-11-17 at 11 04 52 AM" src="https://user-images.githubusercontent.com/45575454/99416033-69e23480-28c6-11eb-9454-c086d4a75997.png">
Error creating duplicate task|<img width="459" alt="Screen Shot 2020-11-17 at 11 05 27 AM" src="https://user-images.githubusercontent.com/45575454/99416040-6c448e80-28c6-11eb-9159-d4012f006ee0.png">

### Code Documentation Updates
